src/natcap/invest/seasonal_water_yield/seasonal_water_yield.py

```diff
@@ -879,133 +875,178 @@ def _calculate_annual_qfi(qfm_path_list, target_qf_path):
         None.
 
     """
-    qf_nodata = -1
-
     def qfi_sum_op(*qf_values):
         """Sum the monthly qfis."""
-        qf_sum = numpy.zeros(qf_values[0].shape)
-        valid_mask = ~utils.array_equals_nodata(qf_values[0], qf_nodata)
```

Comment:
We were assuming that all 12 monthly quickflow rasters would have nodata in the same areas, but that's not necessarily true.

Reply:
Why would that not be true? If a pixel does have no quickflow in a month, I'd expect it to be 0. If its outside of the model domain it would be always nan?